parser/src/parser/grammar.lalrpop

```diff
@@ -345,6 +348,33 @@ ListComprehension<T>: ListComprehension = {
     }
 }
 
+ListFoldingType<T>: ListFoldingType = {
+    "sum" <list_folding: ListFolding<T>> => ListFoldingType::Sum(list_folding),
+    "prod" <list_folding: ListFolding<T>> => ListFoldingType::Prod(list_folding),
+}
+
+ListFolding<T>: ListFolding = {
+    "(" "[" <expr: T> "for" <members: Members> "in" <iterables: Iterables> "]" ")"
+        =>? match (members, iterables) {
+            (Members::Single(..), Iterables::Double(..)) => Err(ParseError::User {
+                error: Error::ParseError(
+                    InvalidListFolding("For a single member, a single iterable is required".to_string())
+                )
+            }),
+            (Members::Single(..), Iterables::Range(..)) => Err(ParseError::User {
+                error: Error::ParseError(
+                    InvalidListFolding("For a single member, a single iterable is required".to_string())
+                )
+            }),
```

Comment:
can these be combined into a single match arm?

Reply:
Thanks. Done.